pkg/scheduler/plugins/numaresource/cpuset.go

```diff
@@ -0,0 +1,318 @@
+/*
+Copyright 2022 The Koordinator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package numaresource
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	maxAvailableCPUCount = 4096
+)
+
+// CPUSetBuilder is a mutable builder for CPUSet. Functions that mutate instances
+// of this type are not thread-safe.
+type CPUSetBuilder struct {
+	result CPUSet
+	done   bool
+}
+
+// NewCPUSetBuilder returns a mutable CPUSet builder.
+func NewCPUSetBuilder() CPUSetBuilder {
+	return CPUSetBuilder{
+		result: CPUSet{
+			elems: map[int]struct{}{},
+		},
+	}
+}
+
+// Add adds the supplied elements to the result. Calling Add after calling
+// Result has no effect.
+func (b CPUSetBuilder) Add(elems ...int) {
+	if b.done {
+		return
+	}
+	for _, elem := range elems {
+		b.result.elems[elem] = struct{}{}
+	}
+}
+
+// Result returns the result CPUSet containing all elements that were
+// previously added to this builder. Subsequent calls to Add have no effect.
+func (b CPUSetBuilder) Result() CPUSet {
+	b.done = true
```

Comment:
modify the copy object is expected behavior?

Reply:
fixed.